tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/contentassist/TclTemplateProposalProvider.xtend

```diff
@@ -56,76 +60,221 @@ class TclTemplateProposalProvider extends DefaultTemplateProposalProvider {
 		super.createTemplates(templateContext, context, acceptor)
 
 		val model = context.currentModel
+		if (model instanceof ComponentTestStepContext) {
+			model?.proposeAvailableInteractions(templateContext, context, acceptor)
+		}
+		if (model instanceof MacroTestStepContext) {
+			model?.proposeAvailableInteractions(templateContext, context, acceptor)
+		}
 		if (model instanceof TestStep) {
 			if (model.hasComponentContext) {
 				val componentContext = model.componentContext
-				componentContext?.proposeAvailableInteractions(model, templateContext, context, acceptor)
-			}else if (model.hasMacroContext){
+				componentContext?.proposeAvailableInteractions(templateContext, context, acceptor)
+			} else if (model.hasMacroContext) {
 				val macro = model.macroContext
-				macro?.proposeAvailableInteractions(model, templateContext, context, acceptor)
+				macro?.proposeAvailableInteractions(templateContext, context, acceptor)
 			}
 		}
 	}
+	
+	/**
+	 * give this' node grammarElement as 'clazz'
```

Comment:
I don't understand that sentence. Maybe something like _returns the provided node instance casted to the provided type, if possible, otherwise returns null_ (?)